frontend_lib/test/component/Timeline.spec.js

```diff
@@ -63,108 +68,195 @@ describe('<Timeline />', () => {
+        describe('when shouldScrollToBottom is set to false', () => {
+          before(() => {
+            scrollIntoViewCallBack.resetHistory()
+            wrapper.setProps({ shouldScrollToBottom: false })
+          })
+          after(() => {
+            wrapper.setProps({ shouldScrollToBottom: true })
+          })
 
-      it('should scroll to the bottom to see the new timline item', () => {
-        expect(scrollIntoViewCallBack.called).to.equal(true)
-      })
-    })
-    describe('a timeline item is added when the timeline scroll is not at the bottom', () => {
-      before(() => {
-        scrollIntoViewCallBack.resetHistory()
-        wrapper.instance().timelineContainer = {
-          scrollTop: scrollTop,
-          clientHeight: clientHeight
-        }
-        wrapper.instance().timelineContainerScrollHeight = scrollTop
-        wrapper.instance().scrollToBottom()
-      })
+          it('should not scroll to the bottom', () => {
+            expect(scrollIntoViewCallBack.called).to.equal(false)
+          })
+        })
+        describe('a timeline item is added when the timeline scroll is at the bottom', () => {
+          before(() => {
+            scrollIntoViewCallBack.resetHistory()
+            wrapper.instance().timelineContainerScrollHeight = scrollTop + clientHeight
+            wrapper.instance().timelineContainer = {
+              scrollTop: scrollTop,
+              clientHeight: clientHeight
+            }
+            wrapper.instance().scrollToBottom(props.timelineData)
+          })
 
-      it('should not scroll to the bottom', () => {
-        expect(scrollIntoViewCallBack.called).to.equal(false)
-      })
-    })
-    describe('a timeline item is added by the current session when the timeline scroll is not at the bottom', () => {
-      before(() => {
-        scrollIntoViewCallBack.resetHistory()
-        wrapper.instance().timelineContainerScrollHeight = scrollTop
-        wrapper.setProps({ isLastTimelineItemCurrentToken: true, newComment: '' })
-        wrapper.instance().timelineContainer = {
-          scrollTop: scrollTop,
-          clientHeight: clientHeight
-        }
-        wrapper.instance().scrollToBottom()
-      })
-      after(() => {
-        wrapper.setProps({ isLastTimelineItemCurrentToken: false, newComment: props.newComment })
-      })
+          it('should scroll to the bottom to see the new timeline item', () => {
+            expect(scrollIntoViewCallBack.called).to.equal(true)
+          })
+        })
+        describe('a timeline item is added when the timeline scroll is not at the bottom', () => {
+          before(() => {
+            scrollIntoViewCallBack.resetHistory()
+            wrapper.instance().timelineContainer = {
+              scrollTop: scrollTop,
+              clientHeight: clientHeight
+            }
+            wrapper.instance().timelineContainerScrollHeight = scrollTop
+            wrapper.instance().scrollToBottom(props.timelineData)
```

Comment:
The `before()` set up the right context and launch the test for the function `.scrollToBottom()`

Reply:
still not clear
why does your context needs to call .scrollToBottom if your test is testing something when the timeline isn't supposed to be at the bottom

Reply:
Before calling `.scrollToBottom` the scroll isn't at the bottom, and after `.scrollToBottom()` was called we need to test if the auto-scroll was enabled because `.scrollToBottom()` has conditions in it (we are testing these conditions in fact).
But maybe I can put `wrapper.instance().scrollToBottom(props.timelineData)` in the `it()` to be more clear

Reply:
Ok that's because scrollToBottom is called by componentDidUpdate.

that's fine then, a bit unclear but consistent